api.planx.uk/modules/webhooks/service/validateInput/schema.ts

```diff
@@ -0,0 +1,50 @@
+import { z } from "zod";
+import { ValidatedRequestHandler } from "../../../../shared/middleware/validate";
+import { isCleanHTML, isObjectValid } from "./utils";
+
+// Definition: https://hasura.io/docs/latest/schema/postgres/input-validations/#response
+type HasuraValidateInputResponse = undefined | { message: string };
+
+// Definition: https://hasura.io/docs/latest/schema/postgres/input-validations/#request
+// Abstract base type that can be merged with specific schemas for data validation
+const hasuraValidateInputRequestSchema = z.object({
+  body: z.object({
+    data: z.object({
+      input: z.array(z.record(z.string(), z.unknown())),
+    }),
+  }),
+});
+
+type HasuraValidateInputRequestSchema = z.infer<
+  typeof hasuraValidateInputRequestSchema
+>;
+
+interface IsCleanJSONBTransformedRequest {
+  body: {
+    isClean: boolean;
+  };
+}
+
+type isCleanJSONBSchema = z.ZodType<
```

Comment:
Using .transform() to join two schemas requires this pretty gnarly type.

Docs: https://zod.dev/?id=zodtype-with-zodeffects